wren-ai-service/eval/prediction.py

```diff
@@ -0,0 +1,138 @@
+import argparse
+import asyncio
+import sys
+import uuid
+from datetime import datetime
+from pathlib import Path
+from typing import Any, Dict, List, Tuple
+
+import orjson
+from langfuse.decorators import langfuse_context, observe
+from tomlkit import document, dumps, parse
+from tqdm import tqdm
+
+sys.path.append(f"{Path().parent.resolve()}")
+import src.utils as utils
+from src.pipelines.ask import generation, retrieval
+from src.pipelines.indexing import indexing
+
+
+def generate_meta() -> Dict[str, Any]:
+    return {
+        "user_id": "wren-evaluator",  # this property is using for langfuse
+        "session_id": f"eval_{uuid.uuid4()}",
+        "date": datetime.now(),
+    }
+
+
+def write_prediction(meta, predictions, dir_path="outputs/predictions") -> None:
+    if Path(dir_path).exists() is False:
+        Path(dir_path).mkdir(parents=True, exist_ok=True)
+
+    output_path = f"{dir_path}/prediction_{meta['session_id']}.toml"
+
+    doc = document()
+    doc.add("meta", meta)
+    doc.add("predictions", predictions)
+
+    with open(output_path, "w") as file:
+        file.write(dumps(doc))
+
+
+def predict(meta: dict, queries: list, pipes: dict) -> List[Dict[str, Any]]:
+    predictions = []
+
+    @observe(name="Prediction Process")
+    async def wrapper(query: dict) -> None:
+        prediction = {
+            "trace_id": langfuse_context.get_current_trace_id(),
+            "trace_url": langfuse_context.get_current_trace_url(),
+            "input": query["question"],
+            "actual_output": [],
```

Comment:
I am thinking whether we need to rename `input`, `actual_output`, `expected_output`, `retrieval_context` and `context`. Since they are not clear enough at first glance

Reply:
discussed offline, we'll add terms into the evaluation framework README